client/src/panels/decompositionPanel.js

```diff
@@ -90,7 +91,7 @@ class DecompositionPanel extends React.Component {
    */
   async fetchDecomposition() {
     let k = 15; // num nearest neighbors to consider when generating M-S complex for a dataset
-    let datasetId = this.props.dataset.datasetId;
+    let datasetId = this.state.datasetId;
```

Comment:
Why was datasetId added to this.state? 
@kylinmb

Reply:
Ultimately, you want each component managing its own data and state; this seems to be the recommendation that React makes. This component was not following that and I cleaned it up where I saw it, but I was not very thorough.